src/rank_llm/rerank/rank_listwise_os_llm.py

```diff
@@ -72,16 +72,19 @@ def run_llm(
     def num_output_tokens(self, current_window_size: Optional[int] = None) -> int:
         if current_window_size is None:
             current_window_size = self._window_size
-        if self._output_token_estimate is None:
-            self._output_token_estimate = (
+        _output_token_estimate = None
+        if self._output_token_estimate is None or self._window_size != current_window_size:
```

Comment:
I think this is going to be wrong after q1, at the beginning of q2, since the last cache value from q1 will be used for the normal window size for q2. The ranker does not get regenerated for each q. 
How about something like this:

if current_window_size == self._window_size:
   if self._output_token_estimate == None:
        self._output_token_estimate = calc
   return self._output_token_estimate
# for different window sizes always calculate:
    return calc

Reply:
the num_output_tokens gets called for each call though, and in this case it does get updated, i did try for this exact case. we return `output_token_estimate if _output_token_estimate` for the last window for q1 and don't update self._output_token_estimate, given that is only filled in the very first case as it is None then and dealing with a full window then.

Reply:
Can you elaborate?

Reply:
ah, thanks for explaining, I missed that you don't update self_output.. in the second case. Your solution works too.
My personal preference is to return per condition. I find it cleaner flow to fallow.
_"at a highlevel if your curr widnow size is the same as standard size, return the starndard num token (calc if needed), else cal and return"_
I find your logic of doing all the calcs and updates first and returining in the end harder to follow. but this is just my personal preference. Since this logic is also correct. Leaving it upto you whether or not you want to update.